pkg/asset/machines/azure/machines.go

```diff
@@ -274,7 +274,7 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 	}
 
 	if platform.CloudName == azure.StackCloud {
-		spec.AvailabilitySet = fmt.Sprintf("%s-cluster", clusterID)
+		spec.AvailabilitySet = fmt.Sprintf("%s_control-plane-as", clusterID)
```

Comment:
Sorry, I might have forgotten/missed why we set it to `"%s_control-plane-as"`? Also, where is this AvailabilitySet being created, now that we switched to CAPI?

Reply:
The availability set is created by CAPZ, so this changes the name to match the capz created AS, hence the unfortunate naming scheme...

Perhaps it makes sense to not include compute nodes in an availability set? But we can keep it this way for now to match what we did in TF.

Reply:
Ah make sense. This gives a pretty good background context for me. Thanks!